js/ui/grid_core/ui.grid_core.header_filter_core.js

```diff
@@ -24,11 +24,17 @@ function resetChildrenItemSelection(items) {
     }
 }
 
-function updateSelectAllState(e, filterValues) {
+function getSelectAllCheckBox(listComponent) {
+    const selector = listComponent.NAME === 'dxTreeView' ? '.dx-treeview-select-all-item' : '.dx-list-select-all-checkbox';
```

Comment:
generate selector using `this.addWidgetPrefix`

Reply:
in this case we would need to use `listComponent._addWidgetPrefix()`. And this is private method, I don't think we need this